source/analysis/AM3DAdditionAB.cpp

```diff
@@ -252,5 +242,29 @@ void AM3DAdditionAB::reviewState()
 	if (valid)
 		setState(0);
 	else
-		setState(AMDataSource::InvalidFlag);
+        setState(AMDataSource::InvalidFlag);
+}
+
+void AM3DAdditionAB::computeCachedValues() const
+{
+    AMnDIndex start = AMnDIndex(0, 0, 0);
+    AMnDIndex end = size();
+    int totalSize = start.totalPointsTo(end);
+
+    QVector<double> data = QVector<double>(totalSize);
+    sources_.at(0)->values(start, end, data.data());
+
+    // Do the first data source separately to initialize the values.
+    cachedData_ = data;
```

Comment:
It is guaranteed since it is only called in `value()` and `values()` and it has to pass the `isValid()` check which does check against empty lists.